### DISCUSSION TITLE
AppMap - A tool to understand how Solidus works

### DISCUSSION THREAD

**Discussion by kgilpin** [solidusio/solidus#3949] [Show and tell]
Greetings, Solidus community. I'm a member of a dev team which is building [AppMap for VSCode](https://marketplace.visualstudio.com/items?itemName=appland.appmap), an extension which automatically generates accurate software design diagrams by running test cases. We've been working extensively with Solidus, and so I am writing this post to make you aware of some resources that you might find helpful to:

* See the design of Solidus in context with the code.
* Trace key code and data flows, and see dependencies as they interact with each other.
* Troubleshoot hard-to-understand bugs using a "top-down" approach.

If you want to try Solidus in your local environment with VS Code, we have a [fork of Solidus](https://github.com/land-of-apps/solidus/tree/appland) that has the AppMap configuration already applied. So, you can skip the setup steps (which are a bit more complicated than usual for Solidus, due to it's multi-gem structure), and go right to mapping.

Here's a video where I step through the process on the `appland` branch: 

<a href="https://www.loom.com/share/baf5b0cb19a24050b1a960a499074eca"> <p>AppMap Solidus in VSCode - Watch Video</p> <img style="max-width:300px;" src="https://cdn.loom.com/sessions/thumbnails/baf5b0cb19a24050b1a960a499074eca-with-play.gif"> </a>

You can also find a complete mapping of Solidus pre-built and available online at https://app.land/land-of-apps/solidus?mapset=2953. This page includes a dependency map of the Solidus code base, trace views of Solidus test cases, a catalog of web service routes, and a basic map of the database schema. Note that on the web site, each "Scenario" corresponds to a test case from Solidus.

If you have questions or want to send us some feedback, you can join our convenient [Discord server](https://discord.com/invite/N9VUap6). Thanks for reading!

Kevin
https://twitter.com/kegilpin
https://dev.to/kgilpin

**Comment by aldesantis** [comment]
Woah, this is some seriously impressive stuff. Thanks for sharing!

**Reply by kgilpin** [reply]
Thanks! Would love to hear more about how you’re using AppMap, how it can be improved, or how you could see this helping the Solidus community.
